webapp/api/advantage.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import flask
```

Comment:
not a big fan of putting `flask` dependencies here.
Can you pass the session object that you have to the function that needs it directly.

Reply:
Leftover import, already taking the session as a parameter(will remove the places where it's still using flask.session by mistake)